### ISSUE TITLE
NixOS option request:  security.pam.u2f.mode and security.pam.u2f.fallback

### ISSUE THREAD

**Issue by Mikilio** [NixOS/nixpkgs#245777]
## Topic
The `pam_u2f.so` library supports the following options (copied from manpage):

```
       userpresence=int
           If 1, require user presence during authentication. If 0, do not request user presence during
           authentication. If omitted, fallback to the authenticator’s default behaviour.

       userverification=int
           If 1, require user verification during authentication (e.g. biometrics). If 0, do not request
           user verification during authentication. If omitted, fallback to the authenticator’s default
           behaviour. If enabled, an authenticator with support for FIDO2 user verification is required.

       pinverification=int
           If 1, require PIN verification during authentication. If 0, do not request PIN verification
           during authentication. If omitted, fallback to the authenticator’s default behaviour. If enabled,
           an authenticator with support for a FIDO2 PIN is required.
```
I'm not quite sure how exactly the options should look like but they should have the ability to express all 8 possible configurations. The fall back option is for the very common use case where you would like to attempt u2f again but in a different mode. This is particularly useful when e.g. unreliable biometric authentication fails.

## Maintainers

NixOS options don't have maintainers so I chose the maintainer of pam-u2f

@philandstuff

**Comment by Majiir** [comment]
There are now (experimental! read [the warning](https://github.com/NixOS/nixpkgs/blob/9d6e6e18bccb8e4dbcc5d68263cdcac16ce51973/nixos/modules/security/pam.nix#L114-L122)) options to do this, introduced in https://github.com/NixOS/nixpkgs/pull/255547. For example:

```nix
{
  security.pam.services.sudo.rules.auth.u2f.settings = {
    userpresence = 1;
    pinverification = 1;
  };
}
```

Currently, these settings must be defined individually for each service (`sudo`, `login`, etc).

**Comment by tomodachi94** [comment]
This appears to be fixed by #255547. Closing...
